src/fsharp/import.fs

```diff
@@ -583,7 +583,7 @@ let ImportILAssemblyTypeForwarders (amap, m, exportedTypes: ILExportedTypesAndFo
                     yield (Array.ofList enc, exportedType.Name), tcref 
                     yield! nested net.Nested (enc @ [ net.Name ]) ]
             yield! nested exportedType.Nested (ns@[n]) 
-    ] |> Map.ofList
+    ] |> dict
```

Comment:
This is the only non-empty dictionary, and I don't know which Equality `dict` uses, but I suspect it's not Structural.
I've sent you a suggestion via DM, as my security software seems to be blocking the code snippet in this comment :(

Reply:
`typeof<string[] * string>.IsValueType` is indeed false, so `dict` didn't do what I thought it would. [source](https://github.com/dotnet/fsharp/blob/2c43ded94bbcd5135e4d594c4a397a9c27796bc2/src/fsharp/FSharp.Core/fslib-extra-pervasives.fs#L161-L171)

Reply:
Having profiled this for the big project I raised https://github.com/dotnet/fsharp/issues/13033 for, I see a reduction in time in `CcuThunk.TryForward` using just `dict`, and no further improvement for the updated version that explicitly uses the HashComparer.

I'll do some benchmarking of the different versions and put them in a comment.